openfisca_core/reforms.py

```diff
@@ -49,120 +61,22 @@ def full_key(self):
-    """
-    Update legislation JSON with a value defined for a specific couple of period defined by
-    its start and stop instant or a period object.
-
-    Returns the modified `legislation_json`.
-
-    This function does not modify its arguments.
-    """
-    assert value is not None
-    if period is not None:
-        assert start is None and stop is None, u'period parameter can\'t be used with start and stop'
-        start = period.start
-        stop = period.stop
-    assert start is not None, u'start must be provided, or period'
-
-    def build_node(node, path_index):
-        if isinstance(node, collections.Sequence):
-            return [
-                build_node(child_node, path_index + 1) if path[path_index] == index else child_node
-                for index, child_node in enumerate(node)
-                ]
-        elif isinstance(node, collections.Mapping):
-            return collections.OrderedDict((
-                (
-                    key,
-                    (
-                        update_items(child_node, start, stop, value)
-                        if path_index == len(path) - 1
-                        else build_node(child_node, path_index + 1)
-                        )
-                    if path[path_index] == key
-                    else child_node
-                    )
-                for key, child_node in node.iteritems()
-                ))
-        else:
-            raise ValueError(u'Unexpected type for node: {!r}'.format(node))
-
-    updated_legislation = build_node(legislation_json, 0)
-    return updated_legislation
-
-
-def update_items(items, start, stop, value):
-    start_str = str(start)
-    stop_str = str(stop.offset(1, 'day')) if stop else None
-
-    i = 0
-    n = len(items)
-    new_items = []
-
-    # Future intervals : not affected
-    if stop_str:
-        while (i < n) and (items[i]['start'] >= stop_str):
-            new_items.append(items[i])
-            i += 1
-
-    # Right-overlapped interval
-    if stop_str:
-        if new_items and (stop_str == new_items[-1]['start']):
-            pass  # such interval is empty
-        else:
-            if i < n:
-                overlapped_value = items[i]['value']
-                new_interval = collections.OrderedDict({'start': stop_str, 'value': overlapped_value})
-                new_items.append(new_interval)
-            else:
-                new_interval = collections.OrderedDict({'start': stop_str, })
-                new_items.append(new_interval)
-
-    # Insert new interval
-    new_interval = collections.OrderedDict({'start': start_str, 'value': value})
-    new_items.append(new_interval)
-
-    # Remove covered intervals
-    while (i < n) and (items[i]['start'] >= start_str):
-        i += 1
-
-    # Past intervals : not affected
-    while i < n:
-        new_items.append(items[i])
-        i += 1
-
-    return new_items
+        assert isinstance(reform_parameters, ParameterNode)
```

Comment:
No. If this is false, there will be no exploitable error message. Don't use assertions.